coordinator/gscoordinator/kubernetes_launcher.py

```diff
@@ -353,13 +425,90 @@ def close_vineyard(self):
         # Use delete deployment instead
         pass
 
-    def create_interactive_instance(self, object_id: int, schema_path: str):
+    def check_if_engine_exist(self, engine_type):
+        """Checks if the engine with the given type exists.
+
+        Args:
+            engine_type: The type of engine to check for.
+
+        Returns:
+            True if the engine exists, False otherwise.
+        """
+
+        engine_pod_name_list = getattr(self, f"_{engine_type}_pod_name_list")
+        engine_pod_ip_list = getattr(self, f"_{engine_type}_pod_ip_list")
+        engine_pod_host_ip_list = getattr(self, f"_{engine_type}_pod_host_ip_list")
+
+        return (
+            engine_pod_name_list is not None
+            and engine_pod_ip_list is not None
+            and engine_pod_host_ip_list is not None
+        )
+
+    def deploy_engine(self, engine_type):
+        """Deploys the engine with the given type.
+
+        Args:
+            engine_type: The type of engine to deploy.
+
+        Returns:
+            A tuple of the pod names, IP addresses, and host IP addresses of the
+            deployed engine.
+        """
+
+        if not self.check_if_engine_exist(engine_type):
+            self._engine_pod_prefix = f"gs-{engine_type}-".replace("_", "-")
+            self._engine_cluster = self._build_engine_cluster(
+                with_analytical_container=engine_type == "analytical",
+                with_analytical_java_container=engine_type == "analytical-java",
+                with_interactive_container=engine_type == "interactive",
+                with_learning_container=engine_type == "learning",
+            )
+            self._create_engine_stateful_set()
+            self._waiting_for_services_ready()
+
+            # Set the engine pod info
+            setattr(self, f"_{engine_type}_pod_name_list", self._pod_name_list)
+            setattr(self, f"_{engine_type}_pod_ip_list", self._pod_ip_list)
+            setattr(self, f"_{engine_type}_pod_host_ip_list", self._pod_host_ip_list)
```

Comment:
Since you use these 3 variables extensively, have you consider extract them to a high level data structure? like a class.